ui/src/reusable_ui/components/button/Button.tsx

```diff
@@ -0,0 +1,96 @@
+// Libraries
+import React, {Component} from 'react'
+import classnames from 'classnames'
+
+// Types
+import {
+  ComponentColor,
+  ComponentSize,
+  ButtonShape,
+  IconFont,
+} from 'src/reusable_ui/types'
+import {RemoteDataState} from 'src/types/flux'
+
+// Styles
+import './Button.scss'
+
+interface Props {
+  text: string
+  onClick?: () => void
+  color?: ComponentColor
+  size?: ComponentSize
+  shape?: ButtonShape
+  icon?: IconFont
+  disabled?: boolean
+  status?: RemoteDataState
+  titleText?: string
+}
+
+class Button extends Component<Props> {
+  public static defaultProps: Partial<Props> = {
+    color: ComponentColor.Default,
+    size: ComponentSize.Small,
+    shape: ButtonShape.Default,
+    disabled: false,
+    status: RemoteDataState.Done,
+  }
+
+  public render() {
+    const {disabled, onClick, text, titleText} = this.props
+
+    return (
+      <button
+        className={this.className}
+        disabled={disabled}
+        onClick={onClick}
+        title={titleText || text}
+      >
+        {this.icon}
+        {this.text}
+        {this.statusIndicator}
```

Comment:
so it would show the status indicator in addition to icon and text? or would it replace those? i think i expected the `render` function to render the various states based on `status`

Reply:
It does! I keep the text & icon in there so the button stays the same size. The text and icon fade out so the spinner doesn't overlap

Reply:
i see.. i wonder if the code can be written in a way that indicates that.. just thinking out loud